infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/tree/SimpleLeafNode.java

```diff
@@ -49,12 +49,10 @@ public Bytes getData() {
 
   @Override
   public Bytes32 hashTreeRoot() {
-    if (hashTreeRoot != null) {
-      return hashTreeRoot;
+    if (cachedHash == null) {
+      cachedHash = Bytes32.wrap(Arrays.copyOf(data.toArrayUnsafe(), MAX_BYTE_SIZE));
     }
-
-    hashTreeRoot = Bytes32.wrap(Arrays.copyOf(data.toArrayUnsafe(), MAX_BYTE_SIZE));
-    return hashTreeRoot;
+    return cachedHash;
```

Comment:
Even if this doesn't wind up as a perf improvement I think I prefer it as it ensures that each execution of the method has a consistent view of cachedHash and not assuming that what it just wrote to the volatile field hasn't changed (not that it would ever change to a bad value with this code anyway).

Reply:
yeah. Going to apply.

Reply:
@ajsutton applied to `SimpleBranchNode` too